File: 0069-sqrtx/0069-sqrtx.py
class Solution(object):
    def mySqrt(self, x):
        if x==1 or x==0:
            return x
        low = 0
        high = x//2
        while low<=high:
            mid = low+(high-low)//2
    
            if mid*mid == x:
                return mid
            elif mid*mid < x:
                low = mid+1
            else:
                high=mid-1
        return high

File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if(x==0 || x==1) return x;
        int l=0, r=(x/2)+1;
        int ans=0;

        while(l<=r){
            long mid = l + (r-l)/2;

            if(mid*mid >x){
                r=mid-1;
            }
            else{
                ans = mid;
                l=mid+1;
            }
        }
        return ans;
    }
};